f1score/mmlu/f1_emb.py

```diff
@@ -0,0 +1,122 @@
+import json
+import torch
+import pandas as pd
+import os
+import random
+
+from sentence_transformers import SentenceTransformer
+
+
+
+def get_rephrase_english_questions(r_path):
+    # get all the questions from the rephrase file
+    questions = []
+    with open(r_path, "r") as fin:
+        for line in fin:
+            tmp_dict = json.loads(line)
+            tmp_dict["text"] = tmp_dict["text"] if tmp_dict["text"] is not None else ""
+            question = tmp_dict["text"].split("\nAnswer:")[0]
+            questions.append(question)
+    return questions
+
+def get_chinese_questions(r_path):
+    # get all the questions from the rephrase file
+    questions = []
+    with open(r_path, "r") as fin:
+        for line in fin:
+            tmp_dict = json.loads(line)
+            question = tmp_dict["text"].split("\n答")[0]
+            questions.append(question)
+    return questions
+
+def get_original_questions(r_path):
+    test_df = pd.read_csv(r_path, header=None)
+    questions = [test_df.iloc[idx, 0] for idx in range(test_df.shape[0])]
+    return questions
+
+
+def compute_f1score(TP, FP, FN):
+    # Compute precision and recall
+    precision = TP / (TP + FP) if (TP + FP) != 0 else 0
+    recall = TP / (TP + FN) if (TP + FN) != 0 else 0
+    
+    # Compute F1 score
+    f1 = 2 * (precision * recall) / (precision + recall) if (precision + recall) != 0 else 0
+    
+    return f1
+
+
+def get_embedding(model, text):
+    query_embedding = torch.tensor(model.encode(text)).cuda().unsqueeze(0).unsqueeze(0)
+    return query_embedding
+
+
+
+
+subjects = ["abstract_algebra", "sociology", "high_school_us_history"]
+
+model = SentenceTransformer('multi-qa-MiniLM-L6-cos-v1')
+
+threshold = 0.5
+
+for subject in subjects:
+    original_questions = get_original_questions(f"data/rephrase/{subject}_test.csv")
+    rephrase_questions = get_rephrase_english_questions(f"data/rephrase/{subject}_test_rephrase_english_filtered_question.jsonl")
+    chinese_questions = get_chinese_questions(f"data/rephrase/{subject}_test_chinese_filtered_question.jsonl")
+
+    original_questions = original_questions[:100]
```

Comment:
why `:100`?

Reply:
Because there are 200 pairs, 100 random pairs and 100 contamination pairs